core/src/types/config.rs

```diff
@@ -198,15 +198,15 @@ impl AppConfig {
         let uproject = UProject::load(&project_path)?;
         Ok(self.get_engine_path(&uproject))
     }
+}
 
-    pub fn ensure_github_pat(&self) -> Result<String, CoreError> {
-        match self.github_pat.clone() {
-            Some(pat) => Ok(pat),
-            None => Err(anyhow!(
-                "GitHub PAT is not configured. Please configure it in the settings.".to_string()
-            )
-            .into()),
-        }
+pub fn ensure_github_pat(github_pat: &Option<String>) -> Result<String, CoreError> {
```

Comment:
this function is a little bit strange now, i feel like we could just remove it and at the callsites just do like 

```rs
let token = config.github_pat.ok_or(CoreError::from.......
```